superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/utils.ts

```diff
@@ -99,3 +100,28 @@ export const doesColumnMatchFilterType = (filterType: string, column: Column) =>
   !column.type_generic ||
   !(filterType in FILTER_SUPPORTED_TYPES) ||
   FILTER_SUPPORTED_TYPES[filterType]?.includes(column.type_generic);
+
+export const mostUsedDataset = (
+  datasets: DatasourcesState,
+  charts: ChartsState,
+) => {
```

Comment:
Your logic here is all good... but for sport, this got me thinking... If you want to get fancy, you might be able to one-liner this with lodash magic... something like:
```
const mostUsedDataset = _(charts)
  .countBy('formData.datasource')
  .toPairs()
  .orderBy([1], ['desc'])
  .value().shift()[0];

return datasets[mostUsedDataset]?.id;
 ```
... but that's probably even more difficult to read 😜

Reply:
Really fancy! Cool functions from Lodash! I think readability is more important though. I had to open Lodash docs to understand how it worked 🤣 .